### PR TITLE
Refactor dbt ls to run from tempdir with symbolic links

### PR DESCRIPTION
**Pull request by tatiana**
Since Cosmos 1.0, `load_method.DBT_LS` is the default dbt project parsing method, unless the user gives a manifest.

Using the original dbt project path has been a source of issues when that path is Read-Only. This issue was faced when running commands that generate `{project-dir}/target/` and `{project-dir}/logs/`, which was solved as part of #414.

This issue is particularly problematic if we want to run `dbt deps` from the original project directory since dbt 1.6 saves adaptors to `{project_dir}/dbt_packages` unless specified in the user's `dbt_project.yml`. To our knowledge, dbt currently does not allow users to define this directory via flags or environment variables, as discussed in #481.

This change aims to solve these issues, by creating a temporary directory and creating symbolic links to the original directory.

Finally, during the development of this task, it was observed that when running dbt ls in a project with `packages.yml`,  dbt raises a 'Compilation Error'. Since dbt may raise other errors in stdout, this PR captures "Errors" more generically - making it more evident potential issues to the end-users.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -115,15 +115,19 @@ jobs:
           DATABRICKS_WAREHOUSE_ID: ${{ secrets.DATABRICKS_WAREHOUSE_ID }}
           DATABRICKS_CLUSTER_ID: ${{ secrets.DATABRICKS_CLUSTER_ID }}
           COSMOS_CONN_POSTGRES_PASSWORD: ${{ secrets.COSMOS_CONN_POSTGRES_PASSWORD }}
+          POSTGRES_HOST: localhost
```
Comment (harels) [inline .github/workflows/test.yml:118]: is this related to this PR?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -180,6 +184,12 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (tatiana) [reply .github/workflows/test.yml:187]: https://github.com/astronomer/astronomer-cosmos/pull/488#discussion_r1308581403

**Review by harels — APPROVED** [review]
